src/python/pants/engine/internals/synthetic_targets_test.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import pytest
+
+from pants.core.target_types import FileTarget, GenericTarget
+from pants.engine.addresses import Address
+from pants.engine.internals.synthetic_targets import (
+    SyntheticTargetAdaptor,
+    SyntheticTargetAdaptors,
+    SyntheticTargetAdaptorsRequest,
+    rules,
+)
+from pants.engine.rules import rule
+from pants.engine.target import (
+    DescriptionField,
+    InvalidTargetException,
+    Tags,
+    WrappedTarget,
+    WrappedTargetRequest,
+)
+from pants.engine.unions import UnionRule
+from pants.testutil.rule_runner import RuleRunner, engine_error
+from pants.util.strutil import softwrap
+
+
+class SyntheticExampleTargetAdaptorsRequest(SyntheticTargetAdaptorsRequest):
+    pass
+
+
+@rule
+async def example_synthetic_targets(
+    _: SyntheticExampleTargetAdaptorsRequest,
```

Comment:
Should this be updated to actually use the path?

It might be good to help enforce the API by actually validating that all of the returned addresses/targets have a matching base-name.

Reply:
Good idea. 👍🏽